### ISSUE TITLE
How can I use FailedThresholdCallback configuration through dependency injection

### ISSUE THREAD

**Issue by usama-azakaw** [dotnetcore/CAP#522]
Once FailedThresholdCallback occurs I want to call a service that Implements an Interface, how this can be done in a Dependency injection way.

**Comment by usama-azakaw** [comment]
The below code that you have provided in the sample is quite basic and can be used only to log in console. But consider I want to send email using IEmailService class how can I do that.

 x.FailedThresholdCallback = (type, msg) =>
                {
                    Console.WriteLine(
                        $@"A message of type {type} failed after executing {x.FailedRetryCount} several times, requiring manual troubleshooting. Message name: {msg.GetName()}");
                };

**Comment by yang-xiaodong** [comment]
Hello , Thanks for your feedback.

Now, you can get the `ServiceProvider` like this:

```cs
x.FailedThresholdCallback = failed =>
{
    var logger = failed.ServiceProvider.GetService<ILogger<Startup>>();
    logger.LogError($@"A message of type {failed.MessageType} failed after executing 
                   {x.FailedRetryCount} several times, requiring manual troubleshooting. 
                   Message name: {failed.Message.GetName()}");
};
```

**Comment by yang-xiaodong** [comment]
Fixed in version `3.0.4-preview-103991021`
